### ISSUE TITLE
Investigating writing a Scraper to catch one of my podcasters

### ISSUE THREAD

**Issue by walt93** [WWBN/AVideo#7668]
**Describe the question**
I have a podcaster, not super tech savvy, he lets Rumble DVR his show.

I want to write a scraper that catches his program & throws it in the encoder queue, waits for it to finish, and builds the updated Roku channel.

I assume I can create an entry in the encoder database and it should wake up and start processing?

Any advice you'd give before I start trying to hack this together? I'm going to write a python script that
 - scrapes Rumble using BeautifulSoup or something,
 - get the latest video, look it up on the streamer database to make sure its new
 - if it's new, throw it in the encoder queue by creating the appropriate entry in the encoder databsase
 - then I have to check on the streamer side - wait for the video to change it's state to "a" (? is this right)
 - once it's gone active, then I need to [Clear Cache Directory] (Help? How do I do this from a python script?)
 - Then I can run my script to build the Roku JSON & upload it to the CDN

**Comment by DanielnetoDotCom** [comment]
> scrapes Rumble using BeautifulSoup or something,

you may want to use the feeds instead a scrap 
https://help.rumble.com/MRSS-&-OTT-Feeds.html

> get the latest video, look it up on the streamer database to make sure its new

the feed may give you some ID, use it somewhere in the AVideo, maybe in the title 

> if it's new, throw it in the encoder queue by creating the appropriate entry in the encoder databsase

Check this...
https://github.com/WWBN/AVideo/wiki/Upload-videos-from-third-party-applications 

you can use a CURL function

> then I have to check on the streamer side - wait for the video to change it's state to "a" (? is this right)
> once it's gone active, then I need to [Clear Cache Directory] (Help? How do I do this from a python script?)

yes you can wait or you can create a custom plugin to trigger when there is a new video, here are 2 different samples

Check this https://github.com/WWBN/AVideo/blob/e7fc048ef75a15e86d99fe8fd579c1c6f4b9f17f/plugin/DiscordNotify/DiscordNotify.php#L45-L120

and this
https://github.com/WWBN/AVideo/blob/e7fc048ef75a15e86d99fe8fd579c1c6f4b9f17f/plugin/VimeoAPI/VimeoAPI.php#L49-L54

you can use this event to call your script

> Then I can run my script to build the Roku JSON & upload it to the CDN

**Comment by walt93** [comment]
I've got the scraper working, but comparing against what the encoder has done previously is failing.

What is the transform that's happening on the title when it goes into the database? I need to use the same transform.

It looks like double quotes have been replaced with two single quotes.

E.g. in the database we have two single quotes in a row:

THE MIDWEEK RIDE with KARLI BONNE'! ep-59 ''Fear Not, We All Were Meant To Witness This Time.''

and my scraper is returning an escaped double quote \"

THE MIDWEEK RIDE with KARLI BONNE'! ep-59 \"Fear Not, We All Were Meant To Witness This Time.\"
